jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/Content.java

```diff
@@ -155,6 +155,22 @@ public static void copy(Source source, Sink sink, Chunk.Processor chunkProcessor
      */
     public interface Source
     {
+        /**
+         * Factory of {@link Content.Source}.
+         */
+        interface Factory
+        {
+            /**
+             * Creates a new {@link Content.Source}.
+             *
+             * @param sizedBufferPool the {@link ByteBufferPool.Sized} to get buffers from. {@code null} means allocate new buffers as needed.
+             * @param first the first byte of the resource to start from.
+             * @param length the length of the content to make available, -1 for the full length.
+             * @return a {@link Content.Source}.
+             */
+            Content.Source newContentSource(ByteBufferPool.Sized sizedBufferPool, long first, long length);
```

Comment:
It would be good to have this convenience method
```suggestion
            Content.Source newContentSource(ByteBufferPool.Sized sizedBufferPool, long first, long length);
            
            default Content.Source newContentSource() { return newContentSource(null, 0L, -1L); }
```

Note that `MultiPart`, which should implement this interface, has a known buffer pool, so the no args would be implemented to use that.

Reply:
I've modified `MultiPart` to make use of that new `Content.Source.Factory` interface, and I could not convince myself that adding that convenience method to `Content.Source.Factory` was worth it.

But it does have some merit on `MultiPart`, so IMHO it's better to have both `newContentSource` methods only there.